src/importer.ts

```diff
@@ -0,0 +1,79 @@
+import {EventMessage} from "./models";
+import {PapiClient} from "./papi-client";
+import {URL} from "./types";
+import * as Promise from "bluebird";
+
+export interface AmqpQueues {
+    event: string;
+}
+
+export interface AmqpConfig {
+    host: URL;
+    queues: AmqpQueues;
+}
+
+export interface ImporterConfig {
+    amqp: AmqpConfig;
+    maxBatchSize: number;
+    maxBatchWaitTimeMs: number;
+    delayBetweenBatchMs: number;
+}
+
+export class Importer {
+
+    constructor(
+        private papiClient: PapiClient,
+        private config: ImporterConfig,
+        private amqpConnectionProvider
+    ) {}
+
+    public importAllEvents(): void {
+        let runCount = 0;
+        this.amqpConnectionProvider
+            .flatMap(connection => connection.createChannel())
```

Comment:
I agree you however it's how the rx-amqplib is built (see https://github.com/SkippyZA/rx-amqplib). I did not investigate about that, but using `map` does not work there.